statsmodels/base/model.py

```diff
@@ -474,6 +474,135 @@ def fit(self, start_params=None, method='newton', maxiter=100,
         return mlefit
 
 
+    def _fit_zeros(self, keep_index=None, start_params=None, return_auxiliary=False, **fit_kwds):
+        """experimental fit the model subject to zero constraints
+
+        Intended for internal use cases until we know what we need.
+        API will need to change to handle models with two exog.
+
+        This is essentially a simplified version of `fit_constrained`, and does not need
+        to use `offset`.
+
+        The estimation creates a new model with transformed design matrix,
+        exog, and converts the results back to the original parameterization.
+
+        Some subclasses could use a more efficient calculation than using a new model.
+
+        Parameters
+        ----------
+        keep_index : array_like (int or bool) or slice
+            variables that should be dropped.
+        start_params : None or array_like
+            starting values for the optimization. `start_params` needs to be
+            given in the original parameter space and are internally
+            transformed.
+        **fit_kwds : keyword arguments
+            fit_kwds are used in the optimization of the transformed model.
+
+        Returns
+        -------
+        results : Results instance
+
+        """
+        # wee need to append index of extra params to keep_index as in NegativeBinomial
+        if hasattr(self, 'k_extra') and self.k_extra > 0:
+            # we cannot change the original, TODO: should we add keep_index_params?
+            keep_index = np.array(keep_index, copy=True)
+            keep_index_p = np.concatenate((keep_index,
+                                           list(range(self.exog.shape[1] + self.k_extra))[-self.k_extra:]))
+        else:
+            keep_index_p = keep_index
+
+        # not all models support start_params, drop if None, hide them in fit_kwds
+        if start_params is not None:
+            fit_kwds['start_params'] = start_params[keep_index_p]
+
+        # build auxiliary model and fit
+        init_kwds = self._get_init_kwds()
+        mod_constr = self.__class__(self.endog, self.exog[:, keep_index], **init_kwds)
+        res_constr = mod_constr.fit(**fit_kwds)
+        #switch name, only need keep_index for params belos
+        keep_index = keep_index_p
+
+        # create dummy results Instance, TODO: wire up properly
```

Comment:
Alternative idea: use `res_constr` which should already be the correct results instance and replace the model,
`res_constr.model = self`
We would loose a functioning results instance of the auxiliary model, but it would be less fragile for recreating the correct instance.